common/tools/esm4mocha.mjs

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { readFile, stat, constants } from "node:fs/promises";
+import { dirname, join } from "node:path";
+import { fileURLToPath } from "url";
+import { Project } from "./dev-tool/node_modules/ts-morph/dist/ts-morph.js";
+
+function consoleLog(...args) {
+  if (process.env.DEBUG && process.env.DEBUG.startsWith("esm4mocha")) {
+    console.log(args);
+  }
+}
+
+function consoleDir(...args) {
+  if (process.env.DEBUG && process.env.DEBUG.startsWith("esm4mocha")) {
+    console.dir(args);
+  }
+}
+
+export async function resolve(specifier, context, defaultResolve) {
+  consoleLog("resolving...");
+  consoleDir({ specifier, context });
+  const resolved = await defaultResolve(specifier, context, defaultResolve);
```

Comment:
Where does defaultResolve come from? Is that part of the `--require` API?

Reply:
the proper name should be `nextResolve` and `nextLoad`. Yes it's part of the loader hooks https://nodejs.org/api/module.html#hooks